### ISSUE TITLE
The template lack deps `@types/react`

### ISSUE THREAD

**Issue by hua03** [react-native-community/react-native-template-typescript#298]
## Expected results

When use other package manager eg: npm, pnpm, no eslint errors are reported.

## Observed results

when use pnpm install new package, there report import error:

```
TS7016: Could not find a declaration file for module 'react'. '/Users/huahao/WorkSpace/Project/Demo/ReactNativeTSApp/node_modules/.pnpm/react@18.1.0/node_modules/react/index.js' implicitly has an 'any' type.   If the 'react' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react'
```

### Logs

no logs, this is a eslint error.

## Steps to reproduce

1. init react native project with typescript template, `pnpx react-native init ReactNativeTSApp --template react-native-template-typescript`
2. install new package with pnpm, `pnpm install @react-navigation/native`

**Comment by DWboutin** [comment]
+1

**Comment by radko93** [comment]
I'm not so familiar with pnpm, do you have any potential solutions for this?

**Comment by hua03** [comment]
I agree with @tom-sherman that `@types/react` should be included in the template rather than being installed by users themselves.

**Comment by emin93** [comment]
Fixed in `6.12.6`, thanks for the contribution @EkaanshArora 👍
